chore: increment versionCode slower [WPB-9816]

vitorhugods opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9816" title="WPB-9816" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-9816</a>  [Android] Simplify VersionName so it's easier to read build names
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Version code is being bumped SO FAST that it is hard to read and has useless numbers to the left. For example the current builds, which are `100000460`. It is gonna take ages until the first 2~3 digits change again, so they're useless.

### Solutions

#### Increase it a bit slower

Every 5min instead of every 10s. That's enough! The automatic bump is to avoid conflict in the PlayStore and we're nowhere close to uploading apps that fast.

In order to do so, modified the code to keep today (21 of June) as the time where it moves from one logic to another, and add 100_001_000 (around the current build number) as a baseline for it.

#### Keep the version name clearer

Instead of showing the _full_ version code, show only the last 5 digits. From 0 to 99_999.

This means, instead of `4.8.0-123456789`, it would be `56789`. MUCH MORE READABLE.

Considering versionCode now bumps every 5min, these are
288 per day
8640 per month
51840 per semester
103_680 per year. ~99_999

So it takes almost a whole year until it rotates back.
It's very unlikely that two APKs with the same version (_e.g._ 4.8.0)
will have the same [leastSignificantVersionCode],
unless they are build almost one year apart.

### Testing

#### Test Coverage

- [X] I have added automated test to this contribution

----
#### PR Post Merge Checklist for internal contributors

- [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

makingthematrix commented on app/src/main/scala/com/waz/zclient/calling/CallingFragment.scala:
```diff
@@ -119,20 +119,22 @@ class CallingFragment extends FragmentHelper {
       if (BuildConfig.MAXIMIZE_MINIMIZE_VIDEO) {
+
+        val isMultiParticipant: Signal[Boolean] = controller.otherParticipants.map(_.size > 2)
+
```
If this is the only place where this signal is used then you could inline it inside the double click function (`controller.otherParticipants.map(_.size > 2).head.foreach { ... }`). But that's ok.

mejdoo replied:
I tried to put this directly below the class, the fragment was crashing.

vitorhugods committed:
chore: increment versionCode slower

In the updated code, the version starts to be incremented every 5 minutes instead of every 10 seconds after June 21, 2024.
Additionally, the version Name is now appended with the leastsignificantVersionCode instead of just the versionCode. The leastsignificantVersionCode is the last 6 digits of the VersionCode.

yamilmedina committed:
Merge branch 'develop' into chore/increment-version-code-slower
